lib/index.js

```diff
@@ -47,218 +45,202 @@ const loginWeb = (opener, opts = {}) => {
 const isValidUrl = u => {
   try {
     return /^https?:$/.test(new URL(u).protocol)
-  } catch (er) {
+  } catch {
     return false
   }
 }
 
-const webAuth = (opener, opts, body) => {
-  const { hostname } = opts
-  body.hostname = hostname || os.hostname()
-  const target = '/-/v1/login'
-  const doneEmitter = new EventEmitter()
-  return fetch(target, {
-    ...opts,
-    method: 'POST',
-    body,
-  }).then(res => {
-    return Promise.all([res, res.json()])
-  }).then(([res, content]) => {
-    const { doneUrl, loginUrl } = content
+const webAuth = async (opener, opts, body) => {
+  try {
+    const res = await fetch('/-/v1/login', {
+      ...opts,
+      method: 'POST',
+      body: {
+        ...body,
+        hostname: opts.hostname || os.hostname(),
```

Comment:
Not sure why I never noticed this, in what universe do we want `os.hostname()` here?